enhancements/olm-deprecate-copied-csvs.md

```diff
@@ -0,0 +1,207 @@
+
+Today, Operator resources are generated on a per-Subscription basis. Since CSVs -- and thus Copied CSVs -- can exist without matching Subscriptions, OLM's controllers will be extended to generate additional Operator resources for these "free CSVs" such that:
+
+- there exists at least one`Operator` resource for every CSV
+- there exists at least one `Operator` resource for every `Subscription`
+
+That is, all CSVs installed by a given `Subscription` share an `Operator` resource __and__ any CSVs that weren't installed by a `Subscription` are related to their own `Operator` resource.
+
+#### Orphaned CSVs
+
+Orphaned CSVs are CSVs that were installed via a `Subscription` that no longer exists. Since the lifetimes -- i.e. the period of existence on a cluster --  of `Operator` resources aren't directly tied to the lifetimes of the resources that triggered their generation, the `Operator` resource(s) associated with Orphaned CSVs will continue to exist _after_ a `Subscription` is deleted. If that `Operator` resource is deleted explicitly, OLM will generate another as per the rules [outlined above](#operator-cardinality).
+
+#### Copied RBAC 
+
+In addition to operator discoverability, Copied CSVs also serve as convienient vehicles for the garbage collection of Copied RBAC. These are resources that are copied along with the CSV to provide the operator access to a namespace. To support the loss of Copied CSVs, the RBAC copying logic in OLM will be extended to write additional owner references from copied resources to the related Operator resource. 
+
+#### The Toggle
+
+A feature toggle will be added to control OLM's use of Copied CSVs. When enabled (toggled on), OLM will delete all existing Copied CSVs and will not generate any more. When disabled (toggled off), OLM will generate Copied CSVs (much like it does today).
+
+When toggled (on then off), OLM will recreate any missing Copied CSVs and reassociate the owner references of related copied resources with them (i.e. the inverse operation of the one described in the [Copied RBAC](#copied-rbac) section).
+
+The toggle will be disabled by default.
+
+The toggle can be set from either a command-line option on the olm-operator binary _or_ a field of the `olm-config` `ConfigMap` colocated in the same namespace as OLM's deployment on-cluster; when both are present, the field of the `ConfigMap` takes precedence.
+
+### Risks and Mitigations
+
+#### Info Leak
+
+**Risk:** Operator status can contain references to resources in namespaces a reader of a Copied CSV wouldn't otherwise be made aware of (be privileged to see); i.e. info leak.
+**Partial Mitigation:** Ensure OLM grants `GET` privileges to specific `Operators` based on their scope as defined by the namespaces the related `OperatorGroup` targets; at minimum, this means namespace admins and editors in these target namespaces.
+
+#### OpenShift Console Integration
+
+**Risk:** The OpenShift Console uses Copied CSVs in the rendering of certain pages.
```

Comment:
Console watches CSVs (including copied CSVs) in many places. How close is the Operator API to the current CSV API? Are all the same data available (CSV annotations, OLM descriptors, status, etc.)?

Reply:
Now that the scope of the enhancement has been reduced to just the toggle -- i.e. no more deprecate/replace -- I don't think we need to worry about this particular question for now.

I've added some new mitigations wrt to console. Please check them out when you get the chance @spadgett.